src/server/utils/wallets/createSmartWallet.ts

```diff
@@ -0,0 +1,152 @@
+import { defineChain, type Address } from "thirdweb";
+import { smartWallet, type Account } from "thirdweb/wallets";
+import { createWalletDetails } from "../../../db/wallets/createWalletDetails";
+import { WalletType } from "../../../schema/wallet";
+import { thirdwebClient } from "../../../utils/sdk";
+import { splitAwsKmsArn } from "./awsKmsArn";
+import {
+  createAwsKmsKey,
+  type CreateAwsKmsWalletParams,
+} from "./createAwsKmsWallet";
+import {
+  createGcpKmsKey,
+  type CreateGcpKmsWalletParams,
+} from "./createGcpKmsWallet";
+import { createLocalWallet } from "./createLocalWallet";
+import { getAwsKmsAccount } from "./getAwsKmsAccount";
+import { getGcpKmsAccount } from "./getGcpKmsAccount";
+
+const getSmartWalletAddresses = async ({
+  adminAccount,
+  accountFactoryAddress,
+}: {
+  adminAccount: Account;
+  accountFactoryAddress?: Address;
+}) => {
+  const smartAccount = smartWallet({
+    chain: defineChain(1),
+    sponsorGas: true,
+    factoryAddress: accountFactoryAddress,
+  });
+
+  const connectedAccount = await smartAccount.connect({
+    client: thirdwebClient,
+    personalAccount: adminAccount,
+  });
+
+  return {
+    accountAddress: connectedAccount.address,
+    accountFactoryAddress,
+  };
```

Comment:
I think this whole function should be adjusted to return a connected account:
```typescript
getSmartAccount = async ({ adminAccount, accountFactoryAddress }): Promise<Account>
```

And have it return a connected smart account.
It's trivial to get the address (`(await getSmartAccount()).address`) and this function can be reused in `walletDetailsToAccount()` below where this code is repeated.